scripts/gogenerate.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+unset -v progdir
+case "${0}" in
+*/*) progdir="${0%/*}";;
+*) progdir=.;;
+esac
+git grep -l '^//go:generate ' | sort -t/ | (
+	gogenerate() {
+		case $# in 0) return;; esac
+		go generate "$@"
+	}
+	unset -v prevdir dir file
+	prevdir=
+	set --
+	while read -r file
+	do
+		case "${file}" in
+		*/*) dir="${file%/*}";;
+		*) dir=".";;
+		esac
+		case "${prevdir}" in
```

Comment:
I am a bit confused on what's the intention of this chunk of code.  Is there another way to write this code that is more readable?

Reply:
Factored the hard part into a separate utility script with comments.  Let me know if it's any better. :-)